A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -2476,59 +2477,90 @@ class FMCMainDisplay extends BaseAirliners {
     }
 
     setPerfApprQNH(s) {
+        if (s === FMCMainDisplay.clrValue) {
+            this.addNewMessage(NXSystemMessages.notAllowed);
+            return false;
+        }
+
         const value = parseFloat(s);
-        const QNH_REGEX = /[0-9]{2}.[0-9]{2}/;
+        const HPA_REGEX = /^[01]?[0-9]{3}$/;
+        const INHG_REGEX = /^[23][0-9]\.?[0-9]{2}$/;
 
-        if (QNH_REGEX.test(value)) {
-            this.perfApprQNH = value;
-            return true;
-        } else if (isFinite(value)) {
-            this.perfApprQNH = value;
-            return true;
+        if (HPA_REGEX.test(value)) {
+            if (value >= 745 && value <= 1050) {
+                this.perfApprQNH = value;
+                return true;
+            } else {
+                this.addNewMessage(NXSystemMessages.entryOutOfRange);
+                return false;
+            }
+        } else if (INHG_REGEX.test(value)) {
+            if (value >= 2200 && value <= 3100) {
+                this.perfApprQNH = value / 100;
+                return true;
+            } else if (value >= 22.0 && value <= 31.00) {
+                this.perfApprQNH = value;
+                return true;
+            } else {
+                this.addNewMessage(NXSystemMessages.entryOutOfRange);
+                return false;
+            }
         }
-        this.addNewMessage(NXSystemMessages.notAllowed);
+        this.addNewMessage(NXSystemMessages.formatError);
         return false;
     }
 
     setPerfApprTemp(s) {
-        const value = parseFloat(s);
-        if (isFinite(value) && value > -270 && value < 150) {
-            this.perfApprTemp = value;
-            return true;
+        if (s === FMCMainDisplay.clrValue) {
+            this.addNewMessage(NXSystemMessages.notAllowed);
+            return false;
         }
-        this.addNewMessage(NXSystemMessages.notAllowed);
-        return false;
+
+        if (!/^[\+\-]?\d{1,2}$/.test(s)) {
+            this.addNewMessage(NXSystemMessages.formatError);
+            return false;
+        }
+        this.perfApprTemp = parseInt(s);
+        return true;
     }
 
     setPerfApprWind(s) {
-        let heading = NaN;
-        let speed = NaN;
-        if (s) {
-            const sSplit = s.split("/");
-            heading = parseFloat(sSplit[0]);
-            speed = parseFloat(sSplit[1]);
+        if (s === FMCMainDisplay.clrValue) {
+            this.addNewMessage(NXSystemMessages.notAllowed);
```

Comment:
I would expect this field to be clearable at all times. We may need pilot confirmation?